RoddyCore/src/de/dkfz/roddy/config/converters/BashConverter.groovy

```diff
@@ -134,81 +134,69 @@ class BashConverter extends ConfigurationConverter {
         ]) {
             if (cfg.getConfigurationValues().getBoolean(bashFlag[0] as String, bashFlag[1] as Boolean)) text << separator << bashFlag[2] as String
         }
-        return text;
+        return text
     }
 
     StringBuilder appendPathVariables() {
-        StringBuilder text = new StringBuilder();
+        StringBuilder text = new StringBuilder()
 
         //Set a path if necessary.
         text << separator << new BashCommandSet().getSetPathCommand()
-        return text;
+        return text
     }
 
     /**
      *
-     * @param cfg    Configuration object. Basically a tree of configuration values that may additionally contain cross-references in the values.
+     * @param cfg Configuration object. Basically a tree of configuration values that may additionally contain cross-references in the values.
      * @return
      */
     Map<String, ConfigurationValue> getConfigurationValuesSortedByDependencies(Configuration cfg) {
-        def values = cfg.getConfigurationValues().getAllValuesAsList();
-        Map<String, ConfigurationValue> listOfUnsortedValues = [:]
-        def listOfSortedValues = new LinkedHashMap<String, ConfigurationValue>();
+        List<ConfigurationValue> values = cfg.getConfigurationValues().getAllValuesAsList()
+        Map<String, ConfigurationValue> listOfUnsortedValues = values.collectEntries { [it.id, it] }
```

Comment:
Can you express somehow the sorting order in the name of the variable? Can you describe the algorithm in a comment? Do I see this correctly that here you generate a list with a partial order on the number of dependencies?